explorer/interpreter/type_checker.cpp

```diff
@@ -1225,10 +1225,20 @@ class TypeChecker::ConstraintTypeBuilder {
   }
 
   // Brings all the `impl`s accumulated so far into the given impl scope.
-  void BringImplsIntoScope(const TypeChecker& type_checker,
-                           Nonnull<ImplScope*> impl_scope) {
-    impl_scope->Add(impl_constraints_, llvm::None, llvm::None, GetSelfWitness(),
+  // Returns the number of impls added, which can be specified as `start_from`
+  // in a subsequent call to add only new `impl`s.
+  auto BringImplsIntoScope(const TypeChecker& type_checker,
+                           Nonnull<ImplScope*> impl_scope,
+                           std::optional<int> start_from = std::nullopt)
```

Comment:
Discussed briefly in person -- I think this semantic would be more suitable to update-in-place with a pointer.

Reply:
Done.